monkey/monkey_island/cc/services/authentication_service/mongo_otp_repository.py

```diff
@@ -26,25 +30,60 @@ def __init__(
     def insert_otp(self, otp: OTP, expiration: float):
         try:
             encrypted_otp = self._encryptor.encrypt(otp.encode())
-            self._otp_collection.insert_one({"otp": encrypted_otp, "expiration_time": expiration})
+            self._otp_collection.insert_one(
+                {"otp": encrypted_otp, "expiration_time": expiration, "used": False}
+            )
+        except Exception as err:
+            raise StorageError(f"Error inserting OTP: {err}")
+
+    def set_used(self, otp: OTP):
+        try:
+            otp_id = self._get_otp_object_id(otp)
+            self._otp_collection.update_one({MONGO_OBJECT_ID_KEY: otp_id}, {"$set": {"used": True}})
+        except UnknownRecordError as err:
+            raise err
         except Exception as err:
-            raise StorageError(f"Error updating otp: {err}")
+            raise StorageError(f"Error updating OTP: {err}")
 
     def get_expiration(self, otp: OTP) -> float:
+        otp_dict = self._get_otp_document(otp)
+        return otp_dict["expiration_time"]
+
+    def otp_is_used(self, otp: OTP) -> bool:
+        otp_dict = self._get_otp_document(otp)
+        return otp_dict["used"]
+
+    def _get_otp_document(self, otp: OTP) -> Mapping[str, Any]:
+        otp_object_id = self._get_otp_object_id(otp)
+        retrieval_error_message = f"Error retrieving OTP {otp} with ID {otp_object_id}"
```

Comment:
Should we be concerned about printing out OTPs?

Reply:
Yes. Good catch.

Reply:
Actually, these are "Secret Varables. See `common.types.otp`. So this wouldn't have been a security violation (yay defensive coding!!!). But it's still better to fix this.